### DISCUSSION TITLE
Delta 2 AC charge switch and sensor

### DISCUSSION THREAD

**Discussion by Katzi1986** [tolwi/hassio-ecoflow-cloud#149] [General]
Request for a feature
------------------------

Hi, I really like your integration, it's working fine with my Delta 2!

However, it would be very usefull, if I could select via Home Assistant, if AC input of the Delta 2 shall be used for charging or not (like in the Ecoflow App under "Lab Features").

Someone else asked for the same feature for the Delta Pro:
https://github.com/tolwi/hassio-ecoflow-cloud/issues/25

Would it possible to add this feature for the Delta 2?

Thanks in advance!

**Comment by raphaelsavina** [comment]
You can use "Prio Solar Charging" to do that. 
I have an automation that use the extra energy from my solar panels with microinverter (so 220V) but when there are not enough Watt (less than 200W), I switch off the AC input by switching on Prio Solar Charging.

**Comment by Katzi1986** [comment]
Thanks for the advice! Indeed, am using the "solar prio switch" the same way and confirm that it basically works. But when "solar prio" is on, and the battery level goes below 20%, it charges with AC power automatically to 70% battery level; Of course this can be avoided with some Home assistant automations, but nevertheless it's not possible to use the last 20% of the battery.
For sure this is not really a big problem, but the a.m. additional feature would solve it.
By the way, a little bit off topic: how do you get out the electrical power from your Delta 2 ? Are you using the DC Output and a grid toe inverter, or the AC Output and connect some devices directly? I tried both, but both were not really satisfying...

**Comment by raphaelsavina** [comment]
Ah, I see, I don't let mine run below 20% :-)

I'm using the AC output which is connected to a switch that let me choose between Delta 2 or grid for one line in my breaker box, this line is for all the lights in the house (220V LED) and sockets for internet access/computers/CCTV and rpi with HA.

**Comment by mhschmall** [comment]
I believe you can specify automatic charge start and stop via the Disharge/ChargeLevel setting. Ecoflow recommends not charging above 85%, which IMHO is stupid.

**Comment by mhschmall** [comment]
I do not find Priority Solar Charging available as a switch from this integration. Is it only available via the ecoflow app?

**Comment by raphaelsavina** [comment]
Maybe you need to do a clean install, it is here:
![Screenshot 2023-03-09 at 00 16 11](https://user-images.githubusercontent.com/540382/223874277-78e27326-d5ce-43e7-9f85-bf5aab948942.png)

**Comment by tyfoon** [comment]
> Thanks for the advice! Indeed, am using the "solar prio switch" the same way and confirm that it basically works.

@Katzi1986 I can't get prio solar charging to work like you have it (=stopping pass through when solar prio is on). I even purchased a official cable for solar charging + a panel connected hoping it would allow me to finally use this. 

Could you share what you are doing? Any particular order to connect things? Do you actually have a panel connected or not? Any other switches on/could you maybe share a screenshot of your settings in homeassistant and/or the app?

Driving me nuts....
